### ISSUE TITLE
Reproducibility

### ISSUE THREAD

**Issue by PG-TUe** [rhiever/tpot#349]
Currently when using TPOT with given `random_state`, not always the same results are obtained.

## Context of the issue
As seen below, invoking TPOT with the same settings and on the same data will not always result in the same pipelines being evaluated. This is troublesome for reproducibility, and makes comparing changes made to TPOT harder to evaluate.

## Process to reproduce the issue

```
from tpot import TPOTClassifier
from sklearn.model_selection import train_test_split
from sklearn.datasets import load_iris

iris = load_iris()
X, y = iris.data, iris.target
X_train, X_test, y_train, y_test = train_test_split(X, y, stratify=y, random_state=42)

for _ in range(5):
    tpot = TPOTClassifier(generations =5, population_size = 10, random_state = 42)
    tpot.fit(X_train,y_train)
    print('score: ', tpot.score(X_test,y_test), ' best pipeline: ', tpot._optimized_pipeline)
```

## Expected result

The same pipeline and accuracy for all 5 runs.

## Current result

score:  0.941538461538  best pipeline:  GaussianNB(input_matrix)
score:  0.941538461538  best pipeline:  KNeighborsClassifier(VarianceThreshold(input_matrix, 0.98999999999999999), 21, 26)
score:  0.941538461538  best pipeline:  ExtraTreesClassifier(SelectPercentile(input_matrix, 48), 6, 0.13)
score:  0.980512820513  best pipeline:  ExtraTreesClassifier(PCA(MinMaxScaler(input_matrix), 36), 39, 0.10000000000000001)
score:  1.0  best pipeline:  GaussianNB(SelectPercentile(CombineDFs(input_matrix, Normalizer(input_matrix, 26)), 34))

## Possible fix

Pull request below.

**Comment by rhiever** [comment]
This should be fixed in the upcoming release (also see #350).
